### ISSUE TITLE
is the Filter2d really faster now?

### ISSUE THREAD

**Issue by AntixK** [kornia/kornia#599]
## 🐛 Bug

PR  #433 mentions that the Filter2d function has been adapted to be faster for larger kernel sizes. Is it really faster though?
I can't seem to reproduce the benchmark results reported in that PR.

## Benchmark Code (Adapted from #433)
```
import torch
import torch.nn.functional as F
import torch.backends.cudnn as cudnn
from time import time
import numpy as np
from tqdm import tqdm
import matplotlib.pyplot as plt

def benchmark_group(input, device, sh):
    torch.cuda.synchronize()
    x = F.conv2d(input, kernel.expand(c,-1,-1,-1), groups=c)
    loss = x.sum()
    loss.backward()
    torch.cuda.synchronize()
    return

def benchmark_reshape(input, device, sh):
    b1,c1,h1,w1 = input.size()
    torch.cuda.synchronize()
    x = F.conv2d(input.view(b*c,1, h1, w1), kernel).view(*sh)
    loss = x.sum()
    loss.backward()
    torch.cuda.synchronize()
    return


def benchmark_adapt(input, device, sh):
    b1,c1,h1,w1 = input.size()
    torch.cuda.synchronize()
    dev = input.device
    ks_numel = kernel.size(3) * kernel.size(2)
    if (ks_numel > 81):
        x = F.conv2d(input.view(b*c,1, h1,w1), kernel).view(*sh)
    else:
        x =  F.conv2d(input, kernel.expand(c,-1,-1,-1), groups=c)
    loss = x.sum()
    loss.backward()
    torch.cuda.synchronize()
    return

#For square kernels
b,c,h,w = 16,16,256, 256
sh = [b,c,h,w]


l1_m, l1_s = [], []
l2_m, l2_s = [], []
l3_m, l3_s = [], []

N = 10
device = 'cuda'
test_range = np.arange(3, 32, 2)
for ks in tqdm(test_range):
    data = torch.rand(b,c,h,w, device=device).float()
    x = F.pad(data,(ks//2, ks//2, ks//2, ks//2))
    b1,c1,h1,w1 = x.size()
    kernel = torch.rand(1,1,ks,ks, device=device, requires_grad=True).float()
    #warmup
    o = F.conv2d(x, kernel.expand(c,-1,-1,-1), groups=c)
    o = F.conv2d(x.view(b*c,1, h1,w1), kernel).view(b,c,h,w)
    torch.cuda.synchronize()

    # print (ks, 'group', str(device))

    l = []
    for _ in range(N):
        s =  time()
        benchmark_group(x, device, sh)
        l.append(time() - s)

    l1_m.append(np.mean(l))
    l1_s.append(np.std(l))

    torch.cuda.synchronize()

    l = []
    for _ in range(N):
        s =  time()
        benchmark_reshape(x,device, sh)
        l.append(time() - s)

    l2_m.append(np.mean(l))
    l2_s.append(np.std(l))
    l = []
    for _ in range(N):
        s = time()
        benchmark_adapt(x,device, sh)
        l.append(time() - s)

    l3_m.append(np.mean(l))
    l3_s.append(np.std(l))


plt.figure()
plt.errorbar(test_range, l1_m, l1_s, lw=2, label='Pad')
plt.errorbar(test_range, l2_m, l2_s, lw=2, label='Reshape')
plt.errorbar(test_range, l3_m, l3_s, lw=2, label='Adaptive')
plt.xlabel('Kernel Size', fontsize=15)
plt.ylabel('Time (s)', fontsize=15)
plt.title('Adaptive Group Conv Cuda test')
plt.grid(True)
plt.legend(fontsize=15)
plt.tight_layout()
plt.show()

```

## Benchmark Results

![cuda_result](https://user-images.githubusercontent.com/14088134/88882160-4b4f2380-d26c-11ea-8c83-0d0904c2452f.png)

From the above results, simple padding seems to be the fastest!

## Environment
 - PyTorch Version (e.g., 1.0): 1.5.0
 - OS: Ubuntu 18.04
 - How you installed PyTorch (`conda`, `pip`, source): pip
 - Python version: 3.8.2
 - CUDA/cuDNN version: 10.2
 - GPU models and configuration: GeForce RTX 2070

**Comment by ducha-aiki** [comment]
@AntixK I have re-run the previous benchmark on the CUDA 10.2 and pytorch 1.5.0 and it seems that you are right.
Thank you!

Would you mind to create PR, which goes back to group-conv only?

**Comment by edgarriba** [comment]
@ducha-aiki is there any way to automate (or semi) this ?

**Comment by ducha-aiki** [comment]
@edgarriba I think, now there is no need to. Our solution was needed because only small kernel group conv was optimized. Now it looks like fixed, so we can just rely on pytorch defaults.
